UnityProject/Assets/Scripts/Systems/Antagonists/Objectives/EscapeObjectives/PreventEveryoneFromEscaping.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+
+namespace Antagonists
+{
+	[CreateAssetMenu(menuName="ScriptableObjects/AntagObjectives/PreventEveryoneFromEscaping")]
+	public class PreventEveryoneFromEscaping : Objective
+	{
+		/// <summary>
+		/// The shuttles that will be checked for this objective
+		/// </summary>
+		private List<EscapeShuttle> ValidShuttles = new List<EscapeShuttle>();
+
+		/// <summary>
+		/// Populate the list of valid escape shuttles
+		/// </summary>
+		protected override void Setup()
+		{
+			ValidShuttles.Add(GameManager.Instance.PrimaryEscapeShuttle);
+		}
+
+		/// <summary>
+		/// Complete if the player is alive and on one of the escape shuttles and shuttle has
+		/// at least one working engine
+		/// </summary>
+		protected override bool CheckCompletion()
+		{
+			return !ValidShuttles.Any( shuttle => shuttle.MatrixInfo != null && CheckForPlayersOnShuttle(shuttle));
+		}
+
+		private bool CheckForPlayersOnShuttle(EscapeShuttle shuttle)
+		{
+			LayerMask layersToCheck = LayerMask.GetMask("Players", "NPC");
+			foreach (Transform trans in shuttle.MatrixInfo.Objects)
+			{
+				if (((1 << trans.gameObject.layer) & layersToCheck) == 0)
+				{
+					return true;
+				}
+			}
+			return false;
```

Comment:
Although unlikely atm if the Ai was aboard the shuttle via a camera it would be counted

Reply:
It could substract the owner of the objective’s transform to avoid that

Reply:
Maybe this works
```cs
foreach (Transform trans in shuttle.MatrixInfo.Objects.Except(new List<Transform>{Owner.transform}))
{
	if (((1 << trans.gameObject.layer) & layersToCheck) == 0)
	{
		return true;
	}
}
```

Reply:
`.Except()` is not a function that exists in `Transform` and I couldn't find anything else that can stop the AI from losing to itself.

Reply:
I thought Objects was a list of transforms, not a transform

Reply:
![image](https://user-images.githubusercontent.com/34368774/124608207-d1a5b980-de6e-11eb-9843-801529fb455b.png)

Reply:
actually wait, can I convert that into a list?

Reply:
apparently not

Reply:
Actually just remembered that Ai is on the ghost layer so will be fine